packages/search/src/utils.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * A `React.useEffect` that will not run on the first render.
```

Comment:
Nit: this is a clear description of what it does, but it doesn't answer _why_ this hook was added/is needed. A short paragraph with an example of when to use it would be nice :-)